### PR TITLE
[BEAM-3326] Remove SdkHarnessClientControlService

### PR DESCRIPTION
**Pull request by tgroh**
The utility of this service is minimal, at best. The actual client
construction is a trivial call to `SdkHarnessClient.create`, and users
can just wrap the lower-level client with that call.

------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/projects/BEAM/issues/) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue.
 - [x] Write a pull request description that is detailed enough to understand:
   - [x] What the pull request does
   - [x] Why it does it
   - [x] How it does it
   - [x] Why this approach
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

### CODE REVIEW

**Comment by tgroh** [comment]
R: @bsidhom

**Comment by bsidhom** [comment]
@axelmagn FYI

**Comment by lukecwik** [comment]
@angoenka Might be interested in these changes since he has been working towards something like this for Dataflow.

File: runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/control/SdkHarnessClientControlService.java [hunk 1]
```diff
@@ -1,75 +0,0 @@
...
-    this.dataService = dataService;
-    activeClients = new ConcurrentLinkedQueue<>();
-    pendingClients = new SynchronousQueue<>();
-    clientPoolService = FnApiControlClientPoolService.offeringClientsToPool(pendingClients);
-  }
-
-  public SdkHarnessClient getClient() {
-    try {
-      // Block until a client is available.
-      FnApiControlClient getClient = pendingClients.take();
```
Comment (bsidhom) [inline runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/control/SdkHarnessClientControlService.java]: The fact that `FnApiControlClientPoolService` requires direct interaction with the underlying client queue makes direct work with the `FnApiControlClientPoolService` difficult to test, as I mentioned in #4751 . I think that this is a compelling enough reason to keep this interface until/if we change the `FnApiControlClientPoolService` interface.

**Comment by tgroh** [comment]
Assuming the changes in https://github.com/apache/beam/pull/4866 are accepted, this likely becomes a no-op including when run against flink, because it removes the need for the `EnvironmentManager` to directly access data or state services.

**Comment by tgroh** [comment]
@bsidhom Is your concern still true?

**Comment by bsidhom** [comment]
No. We can go ahead and remove this now.

**Comment by bsidhom** [comment]
It looks like there are merge conflicts now. Is the plan still to remove this?

**Comment by tgroh** [comment]
run java precommit

**Pull request merged**